### PR TITLE
Add new border-color and background-color utilities

### PR DESCRIPTION
**Pull request by mdo**
Pulling out this commit from #30778 which is outdated and kind of a mess.

- bg-gray is new and maps to gray-400
- border-gray-500 and border-gray-700 are new and are meant to complement bg-gray and bg-secondary as one level off from the borders

Still TBD if we want it, but figured I'd separate this out for now to see if we want/need it.

Preview:

* https://deploy-preview-31694--twbs-bootstrap.netlify.app/docs/5.0/utilities/borders/#border-color
* https://deploy-preview-31694--twbs-bootstrap.netlify.app/docs/5.0/utilities/colors/#background-color

### CODE REVIEW

**Comment by mdo** [comment]
Let me know what you think @twbs/css-review.

**Comment by ffoodd** [comment]
Seems good to me, and useful (I always add a few background utilities).

**Comment by MartijnCuppens** [comment]
> Seems good to me, and useful (I always add a few background utilities).

Agreed.

File: scss/_utilities.scss [hunk 1]
```diff
@@ -100,7 +100,15 @@ $utilities: map-merge(
     "border-color": (
       property: border-color,
       class: border,
-      values: map-merge($theme-colors, ("white": $white))
+      values: map-merge(
+        $theme-colors,
+        (
```
Comment (ffoodd) [inline scss/_utilities.scss:105]: What about using a dedicated variable here, something like `$background-colors` to handle the map and ease customisation?

**Comment by mdo** [comment]
Closing for #34100.

**Pull request closed without merging**